src/call_wrapper.ts

```
@@ -5,24 +5,60 @@ import Module = WebAssembly.Module;
 
 class CallWrapper {
 
-    private static fillMemory(parameters: any[], writeableMemory: Uint32Array) {
-        // A copy of the parameters array is needed, to not override the content of the original one
+    private static fillMemory(memoryLayout: Map<number, number>, memory: WebAssembly.Memory) {
+        const i32Memory = new Uint32Array(memory.buffer);
+        const f64Memory = new Float64Array(memory.buffer);
+
+        for (const entry of memoryLayout.entries()) {
+            const key = entry[0];
+            const value = entry[1];
+
+            if (isOfType(value, WebAssemblyType.INT_32)) {
+                i32Memory[key  / 4] = value;
+            } else if (isOfType(value, WebAssemblyType.FLOAT_64)) {
+                f64Memory[key / 8] = value;
+            }
+        }
+    }
+
+    private static getMemoryLayout(parameters: any[], signature: FunctionSignature): [Map<number, number>, any[]] {
         const fixedParameters = parameters.concat();
-        let index = 0;
+        const memory = new Map<number, number>();
+
+        let wasmMemoryIndex = 0;
+
+        for (let i = 0; i < parameters.length; i++) {
+            const current = parameters[i];
+            const type = signature.parameterTypes[i];
+
+            if (type === WebAssemblyType.INT_32_ARRAY) {
+                memory.set(wasmMemoryIndex, current.length);
```

The code in both branches are the same. You could combine them by moving it out of the branches and add a new variable, that gets set to 4 resp. 8 in the branches.
We should also throw an error in the else part, if a new array type gets introduced.

Something like that:
```javascript
let offset;
if (Int) {
    offset = 4
} else if (Double) {
    if (!aligned) {...}
    offset = 8;
} else {
    throw new Error();
}
// combined logic
```